components/popover/popover-mixin.js

```diff
@@ -1,24 +1,44 @@
 import '../colors/colors.js';
 import '../focus-trap/focus-trap.js';
 import { clearDismissible, setDismissible } from '../../helpers/dismissible.js';
-import { css, html } from 'lit';
+import { css, html, nothing } from 'lit';
 import { getComposedActiveElement, getFirstFocusableDescendant, getPreviousFocusableAncestor } from '../../helpers/focus.js';
-import { isComposedAncestor } from '../../helpers/dom.js';
+import { getComposedParent, isComposedAncestor } from '../../helpers/dom.js';
+import { _offscreenStyleDeclarations } from '../offscreen/offscreen.js';
+import { RtlMixin } from '../../mixins/rtl/rtl-mixin.js';
+import { styleMap } from 'lit/directives/style-map.js';
 
+const pointerLength = 16;
+const pointerRotatedLength = Math.SQRT2 * parseFloat(pointerLength);
 const isSupported = ('popover' in HTMLElement.prototype);
 
 // eslint-disable-next-line no-console
 console.log('Popover', isSupported);
 
-export const PopoverMixin = superclass => class extends superclass {
+export const PopoverMixin = superclass => class extends RtlMixin(superclass) {
```

Comment:
I was hoping we'd be able to EOL/remove `RtlMixin` at some point in favour of logical properties everywhere, but I see this is using it for something else. Wondering if we could maybe introduce something else that doesn't result in a reactive `dir` prop sprouting on the host?

Reply:
Hmmm, it looks like `inset-inline-start` is the replacement for `left`. Likewise, `inset-inline-end` corresponds to `right` when LTR. Browser support is ([CanIUse](https://caniuse.com/?search=inset-inline-start)):
```
Chrome >=87 (legacy)
Safari >= 14.1 (legacy)
Firefox >= 63 (blocked)
```
That's what we'd ideally use, but the positioning behaviour would be completely broken in earlier browsers. It would probably be displayed in top corner, but we maybe it could be set up with fallback to center (the browser's popover default). Maybe we could position using margin logical properties since they are supported by earlier browser versions. But that doesn't feel much better to me.

I'm not confident the position calculations would work as-is. They might need to be revamped, which I think would be "ok". Having just spent a lot of time on the code, I'm fairly current on what they are all doing.

Reply:
I'll send you the new tiers that start in January privately.